#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
from os.path import abspath, basename, dirname, splitext

import math

PACKAGE = basename(dirname(dirname(abspath(__file__))))

gen = ParameterGenerator()

gen.add("scale",                          double_t, 0, "Scale, multiplier for node and edge size", 1.0, 0.0, 1000.0)
gen.add("visualize_nodes",                bool_t,   0, "If true Nodes will be added for Visualization of the roadmap", True)
gen.add("arrow_visualization",            bool_t,   0, "If True Arrows will be used for the Visualization of edges, otherwise Lines are used", True)
gen.add("edge_cost_visualization",        bool_t,   0, "If True the cost on each edge will be added to the Visualization", True)
gen.add("one_direction_only",             bool_t,   0, "If True there will only be one directional marke added between two nodes, be aware of this if you use ARROWS!",False)

exit(gen.generate(PACKAGE, PACKAGE + "_node", splitext(basename(__file__))[0]))
